bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -216,6 +248,26 @@ final class BagsManager(executorService: ExecutorService) {
         case _                             => // Ignore
       }
   }
+
+  /** BAGS status changes. **/
+  def bagsStatus(key: SPNodeKey): Option[BagsStatus] =
+    synchronized(state.statuses.get(key))
+
+  /** Listeners for BAGS status changes. **/
+  private var listeners: List[BagsStatusListener] = Nil
```

Comment:
If I remember correctly you can use `CopyOnWriteArrayList` to hold listeners without synchronization. Maybe @swalker2m has some experience using it?